sql/src/test/java/io/crate/integrationtests/IngestionServiceIntegrationTest.java

```diff
@@ -0,0 +1,234 @@
+            while (produceData) {
+                try {
+                    dataFlowSemaphore.acquire();
+                    Integer nextValue = dataSource.next();
+                    implementation.ingestData(DATA_TOPIC, nextValue);
+                    lastProducedData.set(nextValue);
+                } catch (InterruptedException e) {
+                    // will retry if interrupted
+                } finally {
+                    dataFlowSemaphore.release();
+                }
+                dataFlowingLatch.countDown();
+            }
+        }, "DataIngestionThread");
+        dataIngestionThread.start();
+    }
+
+    @After
+    public void dropIngestRuleAndStopProducingData() throws InterruptedException {
+        execute("drop ingest rule if exists " + UNDER_TEST_INGEST_RULE_NAME);
+        produceData = false;
+        resumeDataIngestion();
+        dataIngestionThread.join(5000);
+    }
+
+    @Test
+    public void testIngestionImplementationIngestsData() throws Exception {
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        pauseDataIngestion();
+
+        execute("select * from ingest_data_raw order by data desc");
+        assertThat(response.rows()[0][0], is(lastProducedData.get()));
+    }
+
+    @Test
+    public void testCreateRuleOnDifferentTopicThanIncomingData() throws Exception {
+        execute("create table other_topic_raw_table (data int)");
+        execute(String.format(Locale.ENGLISH,
+            "create ingest rule rule_for_other_topic on %s where topic = '%s' into %s",
+            INGESTION_SOURCE_ID, "other_topic", "other_topic_raw_table")
+        );
+
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        execute("select * from other_topic_raw_table order by data desc");
+        assertThat(response.rowCount(), is(0L));
+    }
+
+    @Test
+    public void testCreateMultipleRulesForTheSameTargetTable() throws Exception {
+        execute(String.format(Locale.ENGLISH,
+            "create ingest rule another_topic_rule on %s where topic = '%s' into %s",
+            INGESTION_SOURCE_ID, DATA_TOPIC, TARGET_TABLE)
+        );
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        pauseDataIngestion();
+
+        // last produced data should've been ingested twice in the target table
+        execute("select * from ingest_data_raw order by data desc limit 2");
+        assertThat(response.rows()[0][0], is(lastProducedData.get()));
+        assertThat(response.rows()[1][0], is(lastProducedData.get()));
+    }
+
+    @Test
+    public void testDropIngestRuleWillCauseImplementationToStopIngesting() throws Exception {
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        pauseDataIngestion();
+        execute("drop ingest rule " + UNDER_TEST_INGEST_RULE_NAME);
+        int expectedLastIngestedValue = lastProducedData.get();
+        // data that's flowing in after the rule was dropped should not be inserted into the target table
```

Comment:
pause to get the expected value. without that what can we assert on? 
resume having data flow in when we have no rules.

"this does imply somehow that the implementation is not stopping to ingest when the rule is dropped". That's exactly what we are asserting here, that the implementation stopped ingesting - hence assert the last value inserted was the one before the rule was dropped